### DISCUSSION TITLE
wESP32 ethernet support

### DISCUSSION THREAD

**Discussion by eyeson** [arendst/Tasmota#18215] [Ideas]
I have purchased a few wESP32s for use without doing my full due diligence. Turns out since Rev 7+ they have a RTL8201FI rather than the LAN8720AI.
This means currently I can't get the wESP32 to work over ethernet. Everything else works perfectly, a great chip.
I've tried searching to find if anyone (more technical than me) has been able to get it working - but surprisingly I haven't been able to find much on the matter.
https://github.com/arendst/Tasmota/issues/8503 is an old issue discussing the idea of getting ethernet working on ESP32, using the wESP as an early example - but from what I can gather it doesn't help me with my problem.
Googling has provided me with this insight https://github.com/esphome/feature-requests/issues/1462 (which is the same request but for ESPHome)
Their big issue appeared to be making sure the framework was at the right version but from my (technically limited) point of view it appears that Tasmota is already at the correct version to be able to support a RTL8201FI ethernet chip.

Is this something that is likely to be supported?

**Comment by s-hadinger** [comment]
I'm surprised it doesn't work. Tasmota already supports RTL8201 with `EthType 2`

See here https://github.com/arendst/Tasmota/blob/9a3a479c4f20197f6836582860475f7534d62fc3/tasmota/tasmota_xdrv_driver/xdrv_82_esp32_ethernet.ino#L74

**Reply by eyeson** [reply]
In which case - I, firstly apologise, and secondly have probably been configuring it wrong.

In my user_config_override.h I have the following:
`#define ETH_TYPE 2
#define EthAddress 0
`

I am using TasmoCompiler (https://github.com/rochowiak86/tasmocompiler) to compile it and I have selected ESP32 generic, and made sure the ethernet option is ticked.

Have I missed something obvious?

**Reply by s-hadinger** [reply]
You don't need to self compile. Just use the standard binary and type `EthType 2` in the console. You also need the other parameters right : `ethaddress` and `ethclockmode`

**Reply by eyeson** [reply]
I have a couple of MCP23017 that I want to use, this drove me down the self compile route.

Also why does EthType 2 means something different here https://tasmota.github.io/docs/Commands/#ethernet?

**Reply by s-hadinger** [reply]
Good catch. The documentation is outdated. Let me update it now.

**Reply by eyeson** [reply]
Thank you - I am still struggling to get this working - I did try the pre built binary and still had no joy. I am pretty sure this is me missing something at this point.

If I continue down the self compiled route , this what my user_config_override.h now looks like

`
#define ETH_ADDRESS 0
#define ETH_CLKMODE 0
#define ETH_TYPE 2
`
Is there a way to define GPIO16 = ETH MDC and GPIO17 = ETH MDIO in the user_config_override.h file?

**Reply by sfromis** [reply]
You can do that by including a template in `USER_TEMPLATE`

**Reply by gon0** [reply]
Hello everyone, I am trying to get wESP32 Rev.7 to run with tasmota

to summarize the steps I understood and tried:
1. Flash the standard image "Tasmota (english)" via https://tasmota.github.io/install/ (it worked, ESP32 is accessible via WiFi)
2. adding the template from https://tasmota.github.io/docs/ESP32-Devices/
` {"NAME":"wESP32","GPIO":[0,0,1,0,1,1,0,0,1,1,1,1,5568,5600,1,0,0,0,0,1,0,0,0,0,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}`
3. according to https://tasmota.github.io/docs/Commands/#ethernet setting
4. EthType 2
5. EthAddress 0
6. EthClockMode 0
7. Ethernet 1 (is already enabled via default)

if I then disconnect the programming tool and plug in the wESP32 into a PoE port, the PoE switch LED is blinking, but not stable on. This seems to indicate, there is "something". On the wESP32 the ethernet LEDs are off.

Which steps or setting may I have missed?
